src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs

```diff
@@ -200,11 +200,10 @@ public override void OnException(Activity activity, object payload)
                         {
                             case SocketError.HostNotFound:
                                 activity.SetStatus(Status.Error.WithDescription(exc.Message));
-                                return;
+                                break;
                         }
                     }
-
-                    if (exc.InnerException != null)
+                    else if (exc.InnerException != null)
                     {
                         activity.SetStatus(Status.Error.WithDescription(exc.Message));
```

Comment:
Not related to this PR really.. but this is suspect. Did we mean to set the status description with the InnerException message? _And_ do we really want to be grabbing the innermost exception here?

Reply:
good catch! I think its the outermost we should use for status message. we can address this separately.